### PR TITLE
Hub reporting - changes for beta 2

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

Implementation of hub reporting RFC https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00085-hub-reporting.md

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/SystemReport_queries.xml [hunk 1]
```diff
@@ -83,34 +83,53 @@
...
+<mode name="SystemNetAddressV4" class="">
+    <query params="offset, limit">
+          SELECT rhnservernetinterface.server_id AS system_id
+                    , rhnservernetinterface.id AS interface_id
+                    , rhnservernetaddress4.address
+                    , rhnservernetaddress4.netmask
+                    , rhnservernetaddress4.broadcast
+            FROM rhnservernetinterface
+                    INNER JOIN rhnservernetaddress4 ON rhnservernetinterface.id = rhnservernetaddress4.interface_id
+        order by system_id, interface_id, address OFFSET :offset LIMIT :limit
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/db/datasource/xml/SystemReport_queries.xml:112]: you uppercased keywords for other queries, so I think you should do it here as well.

**Review by mcalmer — CHANGES_REQUESTED** [review]
For all the changes in the reportdb schema, we need schema migrations.

**Review by lucidd — APPROVED** [review]

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/SystemReport_queries.xml (hunk 1 above)
Comment (mackdk) [reply java/code/src/com/redhat/rhn/common/db/datasource/xml/SystemReport_queries.xml]: Fixed

**Comment by mackdk** [comment]
> For all the changes in the reportdb schema, we need schema migrations.

I have added the upgrade scripts. We definitely need the job to check that everything is fine though.

**Review by mcalmer — APPROVED** [review]

**Pull request merged**